lib/qiita/markdown/filters/inline_code_color.rb

```diff
@@ -0,0 +1,43 @@
+module Qiita
+  module Markdown
+    module Filters
+      class InlineCodeColor < HTML::Pipeline::Filter
+        DEFAULT_CLASS_NAME = "inline-code-color".freeze
+
+        REGEXPS = Regexp.union(
+          /\#(?:\h{3}|\h{6})/,
+          /rgba?\(\s*(?:\d+(?:\,|\s)\s*){2}\d+\s*\)/,
+          /rgba?\(\s*(?:\d+%(?:\,|\s)\s*){2}\d+%\s*\)/,
+          /rgba?\(\s*(?:\d+\,\s*){3}\d*\.?\d+%?\s*\)/,
+          /rgba?\(\s*(?:\d+\s*){2}\d+\s*\/\s*\d?\.?\d+%?\s*\)/,
+          /rgba?\(\s*(?:\d+%\s*){2}\d+%\s*\/\s*\d?\.?\d+%?\s*\)/,
+          /hsla?\(\s*\d+(?:deg|rad|grad|turn)?\,\s*\d+%\,\s*\d+%\s*\)/,
+          /hsla?\(\s*\d+(?:deg|rad|grad|turn)?\s+\d+%\s+\d+%\s*\)/,
+          /hsla?\(\s*\d+(?:deg|rad|grad|turn)?\,\s*(?:\d+%\,\s*){2}\d?\.?\d+%?\s*\)/,
+          /hsla?\(\s*\d+(?:deg|rad|grad|turn)?\s+\d+%\s+\d+%\s*\/\s*\d?\.?\d+%?\s*\)/,
+        )
+
+        COLOR_CODE_PATTERN = /\A\s*(#{REGEXPS})\s*\z/
+
+        def call
+          doc.search(".//code").each do |node|
+            if (color = node.inner_text) =~ COLOR_CODE_PATTERN
+              node.add_child(color_element(color))
```

Comment:
Since `COLOR_CODE_PATTERN` allows leading and trailing whitespaces around the color pattern, I think it should be stripped.

```rb
node.add_child(color_element(color.strip))
```